cs/src/core/io/safe/OutputBuffer.cs

```diff
@@ -220,9 +220,14 @@ public virtual void WriteString(Encoding encoding, string value, int size)
         // Grow the buffer so that there is enough space to write 'count' bytes
         internal virtual void Grow(int count)
         {
-            var minLength = position + count;
+            int minLength = position + count;
             length += length >> 1;
-            if (length < minLength) length = minLength;
+
+            const int ArrayIndexMaxValue = 0x7FFFFFC7;
+            if ((uint)length > ArrayIndexMaxValue)
```

Comment:
Minor: The Bond code base's style is to use braces for all multi-line conditions. Can you reformat this?

```suggestion
            if ((uint)length > ArrayIndexMaxValue)
            {
                length = ArrayIndexMaxValue;
            }
            else if (length < minLength)
            {
                length = minLength;
            }
```